pkg/readers/container_image_resolve_workflow.go

```diff
@@ -0,0 +1,140 @@
+	targetImageId, err := c.findLocalDockerImageId(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// no image found, go to the next workflow
+	if targetImageId == "" {
+		return nil, imageResolverUnsupportedError
+	}
+
+	tempTarFileName, err := c.saveDockerImageToTempFile(ctx, targetImageId)
+	if err != nil {
+		return nil, err
+	}
+
+	image, err := scalibrlayerimage.FromTarball(tempTarFileName, scalibrlayerimage.DefaultConfig())
+
+	if err != nil {
+		return nil, err
+	}
+
+	if err := os.Remove(tempTarFileName); err != nil {
+		return nil, err
+	}
+
+	logger.Infof("using image form local docker image catalog")
+	return image, nil
+}
+
+func (c containerImageReader) imageFromLocalTarFolder(_ context.Context) (*scalibrlayerimage.Image, error) {
+	pathExists, err := checkPathExists(c.imageTarget.imageStr)
+	if err != nil {
+		// Permission denied etc.
+		return nil, err
+	}
+
+	if !pathExists {
+		return nil, imageResolverUnsupportedError
+	}
+
+	containerImage, err := scalibrlayerimage.FromTarball(c.imageTarget.imageStr, scalibrlayerimage.DefaultConfig())
+	if err != nil {
+		return nil, err
+	}
+
+	logger.Infof("using image form tarball")
+	return containerImage, nil
+}
+
+func (c containerImageReader) imageFromRemoteRegistry(_ context.Context) (*scalibrlayerimage.Image, error) {
+	if !c.config.RemoteImageFetch {
+		return nil, fmt.Errorf("remote image fetching is disabled")
+	}
+
+	containerImage, err := scalibrlayerimage.FromRemoteName(c.imageTarget.imageStr, scalibrlayerimage.DefaultConfig())
+	if err != nil {
+		return nil, err
+	}
+
+	logger.Infof("using image form remote registry")
+	return containerImage, nil
+}
+
+func (c containerImageReader) findLocalDockerImageId(ctx context.Context) (string, error) {
+	allLocalImages, err := c.dockerClient.ImageList(ctx, image.ListOptions{})
+	if err != nil {
+		return "", imageResolverUnsupportedError
+	}
+
+	for _, image := range allLocalImages {
+		if slices.Contains(image.RepoTags, c.imageTarget.imageStr) {
+			return image.ID, nil
+		}
+	}
+
+	// no image, without error while finding
+	return "", nil
+}
+
+func (c containerImageReader) saveDockerImageToTempFile(ctx context.Context, targetImageId string) (string, error) {
+	reader, err := c.dockerClient.ImageSave(ctx, []string{targetImageId})
+	if err != nil {
+		return "", imageResolverUnsupportedError
+	}
+
+	// create tem directory in /tmp for storing `POSIX tar archive` in file
+	tempTarFile, err := os.CreateTemp(os.TempDir(), fmt.Sprintf("image-%s-*.tar", c.imageTarget.imageStr))
+
+	if err != nil {
+		return "", err
+	}
+
+	if _, err := io.Copy(tempTarFile, reader); err != nil {
+		return "", err
+	}
+
+	if err := reader.Close(); err != nil {
+		return "", err
+	}
+
+	if err := tempTarFile.Close(); err != nil {
+		return "", err
+	}
+
+	// from docs: it is safe to call Name after Close
+	return tempTarFile.Name(), nil
+}
+
+func checkPathExists(path string) (bool, error) {
```

Comment:
We should have a unit test for this function. Also why is this outside the reader and at the package level? Do we need this function across different readers? If not, I think it should be within the `containerImageReader` struct